### ISSUE TITLE
Database / upgrade issues 5.3.2 to 5.3.5

### ISSUE THREAD

**Issue by yossiarieli532** [LuatixHQ/opencti#2153]
## Description

Keep getting the same error. I tried to simply delete the damaged artifact but it didn't seem to work.

`{"category":"APP","error":{"context":{"category":"technical","error":{"_error":{},"_showLocations":false,"_showPath":false,"_stack":"DatabaseError: A database error has occurred\n    at error (/opt/opencti/build/src/config/errors.js:8:10)\n    at DatabaseError (/opt/opencti/build/src/config/errors.js:40:48)\n    at loadFile (/opt/opencti/build/src/database/minio.js:119:11)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Promise.all (index 28231)\n    at N3r.up (/opt/opencti/build/src/migrations/1652125339035-files_database.js:11:17)","data":{"category":"technical","filename":"import/Artifact/fe5c56bd-a963-45b0-89f6-909e713c5042/vbc.exe","http_status":500,"reason":"File not found","user_id":"6a4b11e1-90ca-4e42-ba42-db7bc7f7d505"},"internalData":{},"name":"DatabaseError","time_thrown":"2022-06-12T04:43:50.329Z"},"http_status":500,"reason":"[OPENCTI] Platform initialization fail"},"message":"An unknown error has occurred","name":"UnknownError","stack":"UnknownError: An unknown error has occurred\n    at error (/opt/opencti/build/src/config/errors.js:8:10)\n    at UnknownError (/opt/opencti/build/src/config/errors.js:54:47)\n    at platformInit (/opt/opencti/build/src/initialization.js:335:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at boot (/opt/opencti/build/src/boot.js:14:5)"},"level":"error","message":"[OPENCTI] Platform start fail","timestamp":"2022-06-12T04:43:51.735Z","version":"5.3.5"}
`

## Environment

1. OS (where OpenCTI server runs): ubuntu 20
2. OpenCTI version: openCTI 5.3.5
3. OpenCTI client: docker
4. Other environment details:

## Reproducible Steps

Upgraded docker-compose.yml to the latest version.

**Comment by richard-julien** [comment]
Hi @yossiarieli532 ,

Can you check if 
1. The file import/Artifact/fe5c56bd-a963-45b0-89f6-909e713c5042/vbc.exe is visible on minio UI.
2. If the file exists on the filesystem of the minio container.

Thanks.

**Comment by yossiarieli532** [comment]
Hi @richard-julien,

Thanks for the quick turnaround! I'm unable to access to minio UI. I keep getting the same error message:
`<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AllAccessDisabled</Code><Message>All access to this resource has been disabled.</Message><Resource>/minio/health/</Resource><RequestId></RequestId><HostId>1fdda636-b554-49af-b2a3-397272b58873</HostId></Error>
`

I also tried via the console but that didn't work either. I'm using the the following image - minio/minio:RELEASE.2022-05-19T18-20-59Z


`The minio docker logs show the following:

You are running an older version of MinIO released 2 weeks ago 
 Update: Run mc admin update

Finished loading IAM sub-system (took 0.0s of 0.0s to load data).
API: http://172.18.0.19:9000  http://127.0.0.1:9000 

Console: http://172.18.0.19:40971 http://127.0.0.1:40971 

Documentation: https://docs.min.io

WARNING: Console endpoint is listening on a dynamic port (40971), please use --console-address ":PORT" to choose a static port.
Exiting on signal: TERMINATED`


Appreciate the help!

**Comment by SamuelHassine** [comment]
Hello @yossiarieli532,

Are you able to browse the filesystem on the MinIO volume / directory? The files are directly accessible. 

Basically, the error you have means that the file is in MinIO but not in OpenCTI, you have to find a way to delete it on MinIO before running the platform. It can be done whether in the MinIO UI or directly in the filesystem.

**Comment by yossiarieli532** [comment]
Hi,

Yes I can. But when I try the mc command I get "sh: mc: command not found"
I'm inside the minio docker container but just can't delete the file...
Any idea what I'm doing wrong?

**Comment by yossiarieli532** [comment]
I manually deleted it and restarted everything but now I keep the same error just a different file...

`{"category":"APP","error":{"context":{"category":"technical","filename":"import/Artifact/fe5ff010-e531-45e9-82e0-446128214769/Transfer application form.exe","http_status":500,"reason":"File not found","user_id":"6a4b11e1-90ca-4e42-ba42-db7bc7f7d505"},"message":"A database error has occurred","name":"DatabaseError","stack":"DatabaseError: A database error has occurred\n    at error (/opt/opencti/build/src/config/errors.js:8:10)\n    at DatabaseError (/opt/opencti/build/src/config/errors.js:40:48)\n    at loadFile (/opt/opencti/build/src/database/minio.js:119:11)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Promise.all (index 28231)\n    at N3r.up (/opt/opencti/build/src/migrations/1652125339035-files_database.js:11:17)"},"level":"error","message":"[MIGRATION] Error during migration","timestamp":"2022-06-12T11:13:26.030Z","version":"5.3.5"}
`

As soon as I deleted that file too the error moved to a different file.

`{"category":"APP","error":{"context":{"category":"technical","error":{"_error":{},"_showLocations":false,"_showPath":false,"_stack":"DatabaseError: A database error has occurred\n    at error (/opt/opencti/build/src/config/errors.js:8:10)\n    at DatabaseError (/opt/opencti/build/src/config/errors.js:40:48)\n    at loadFile (/opt/opencti/build/src/database/minio.js:119:11)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Promise.all (index 28231)\n    at N3r.up (/opt/opencti/build/src/migrations/1652125339035-files_database.js:11:17)","data":{"category":"technical","filename":"import/Artifact/fe612f52-2c24-455c-8491-a19594659521/c0abec41705b4bdd837daf0df85f53ca.exe","http_status":500,"reason":"File not found","user_id":"6a4b11e1-90ca-4e42-ba42-db7bc7f7d505"},"internalData":{},"name":"DatabaseError","time_thrown":"2022-06-12T11:17:21.482Z"},"http_status":500,"reason":"[OPENCTI] Platform initialization fail"},"message":"An unknown error has occurred","name":"UnknownError","stack":"UnknownError: An unknown error has occurred\n    at error (/opt/opencti/build/src/config/errors.js:8:10)\n    at UnknownError (/opt/opencti/build/src/config/errors.js:54:47)\n    at platformInit (/opt/opencti/build/src/initialization.js:335:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at boot (/opt/opencti/build/src/boot.js:14:5)"},"level":"error","message":"[OPENCTI] Platform start fail","timestamp":"2022-06-12T11:17:22.792Z","version":"5.3.5"}
`

**Comment by richard-julien** [comment]
`Basically, the error you have means that the file is in MinIO but not in OpenCTI, you have to find a way to delete it on MinIO before running the platform. It can be done whether in the MinIO UI or directly in the filesystem.`
In fact its not exactly that.

The Minio API is able to list the files but are not able to really access the metadata of the files. If you are not able to access the minio UI it looks like your minio is not in a good shape.

I dont know how this situation can happen. First thing to do i think its to have a minio where you access the UI and see if you have access to the files this way. If you have access try to download it with the minio UI. If you have problem, try to upload a file through the UI and then check the difference between  the uploaded file and "import/Artifact/fe612f52-2c24-455c-8491-a19594659521/c0abec41705b4bdd837daf0df85f53ca.exe" (access rights, etc). If any difference are observed try to realign the existing files with what you see

**Comment by yossiarieli532** [comment]
Very weird but understood. Just to make sure I'm not doing anything stupid, I should be able to access the UI simply by browsing the VM IP on port 9000 correct?

Can I just copy the database, delete everything and just import it back to a clear machine?

**Comment by yossiarieli532** [comment]
Also, it this a normal minio docker output?

'You are running an older version of MinIO released 2 weeks ago   Update: Run `mc admin update`  Finished loading IAM sub-system (took 0.0s of 0.0s to load data). API: http://172.18.0.19:9000  http://127.0.0.1:9000  Console: http://172.18.0.19:40971 http://127.0.0.1:40971  Documentation: https://docs.min.io WARNING: Console endpoint is listening on a dynamic port (40971), please use --console-address ":PORT" to choose a static port. Exiting on signal: TERMINATED`

**Comment by yossiarieli532** [comment]
I updated minio to the latest version. Firewall ports are all open. Still getting access denied.

<Error><Code>AllAccessDisabled</Code><Message>All access to this resource has been disabled.</Message><Resource>/minio/</Resource><RequestId></RequestId><HostId>1fdda636-b554-49af-b2a3-397272b58873</HostId></Error>

Any ideas?

**Comment by richard-julien** [comment]
Really not sure. Looks like Minio process cant access its own files.
Can you check the access rights of the "minio" folder and sub folders ?

**Comment by yossiarieli532** [comment]
When I access the minio container with "docker exec -it docker_minio_1 sh"
I only have the data folder with opencti-bucket and the rest of the subfolders...

But already chmod 755 -R *

**Comment by richard-julien** [comment]
Sorry to not be able to help you more on that. 
Seems related to minio container and user access inside it.
Maybe you can try to contact the minio team to firstly understand why you cant access to the minio UI.

**Comment by yossiarieli532** [comment]
Thanks appreciate the help. I'll reach out.

**Comment by alice-debra** [comment]
Hello @yossiarieli532,
Due to a lack of recent activity on this issue, we will proceed to close it for now. We sincerely thank you for your feedback, and please don’t hesitate to reopen the issue if the problem persists.
Kind regards,
Alice
